### ISSUE TITLE
Feature Request: Integration with project.el

### ISSUE THREAD

**Issue by arkhan** [manateelazycat/lsp-bridge#174]
Excellent work,

Support for [project.el](https://elpa.gnu.org/packages/project.html) can be added, to detect the root of the project

Thank you very much

**Comment by manateelazycat** [comment]
No, project.el can't detect root of project correctly when you use nested git project.

The right way is `git rev-parse --is-inside-work-tree` and `git rev-parse --show-toplevel`

**Comment by arkhan** [comment]
> 

can be extended to behave similar to projectile: https://manueluberti.eu/emacs/2020/11/14/extending-project/

**Comment by manateelazycat** [comment]
> > 
> 
> can be extended to behave similar to projectile: https://manueluberti.eu/emacs/2020/11/14/extending-project/

No, It's too slow scan project files in elisp side, some javascript project has 10,000 files in node_modules directory.
